chordspy/connection.py

```
@@ -0,0 +1,681 @@
+                while manager.running:                 # Main execution loop
+                    time.sleep(1)                      # Prevent CPU overutilization
+                    
+        # WiFi Protocol Handling        
+        elif args.protocol == 'wifi':
+            if manager.connect_wifi():                 # Attempt WiFi connection
+                while manager.running:                 # Main execution loop
+                    time.sleep(1)                      # Prevent CPU overutilization
+                    
+        # BLE Protocol Handling
+        elif args.protocol == 'ble':
+            if manager.connect_ble(args.ble_address):  # Attempt BLE connection
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)